errands/widgets/task_list.py

```diff
@@ -63,22 +76,45 @@ def build_ui(self):
             sensitive=False,
         )
         self.scroll_up_btn.connect("clicked", lambda *_: scroll(self.scrl, False))
+        self.left_toggle_sidebar_bin = Adw.Bin(
+            child=self.left_toggle_sidebar_btn
+        )
+        self.right_toggle_sidebar_bin = Adw.Bin(
+            child=self.right_toggle_sidebar_btn
+        )
```

Comment:
I wanted to ask, why we need Adw.Bin? Why not use buttons directly?

Reply:
I've solved it that way because the button's visibility depends on whether the toolbar is in the top header bar and the preference for the sidebar position. If I'd bind both directly to the button's `visible` property, changing the window size would make the wrong button appear as well even though this button should not appear. Using a bin, both factors have an own visibility to control and this problem does not appear. See lines 228 - 233:
```python
GSettings.bind("right-sidebar", self.left_toggle_sidebar_bin, "visible", True)
# ...
GSettings.bind("right-sidebar", self.right_toggle_sidebar_bin, "visible")
# ...
tasks_brb_bp.add_setter(self.left_toggle_sidebar_btn, "visible", False)
tasks_brb_bp.add_setter(self.right_toggle_sidebar_btn, "visible", False)
```

Maybe there's are more elegant solution, but that's the easiest that came to my mind.